common/workunit/CMakeLists.txt

```diff
@@ -77,6 +77,7 @@ if(NOT PLUGIN)
         nbcd 
         eclrtl 
         deftype
+        thorhelper
```

Comment:
I think this can be avoided, better not to propagate the engine library thorhelper into workunit if it can be avoided.

This is being included because of IndirectCodeContext ? 
Apart from 2 virtuals it actually implements differently, a vanilla IndirectCodeContext could be moved out of thorhelper.
And the version that is needed by the engines (by thorcommon) could derive from the generic IndirectCodeContext and overload the 2 methods.